In [27]:
import wikipedia

# Search forChildrens Authors
First we'll search for the term childrens author and see if there is a good starting page

In [2]:
result = wikipedia.search('Childrens Author')

Result shows that the first page is childrens literature, then a bunch of authors. Lets pull the childrens literature page and see if we can use it as a starting point

In [3]:
result

["Children's literature",
 'Sam Angus (writer)',
 "Alvin Schwartz (children's author)",
 "Douglas Evans (children's author)",
 "Ruth White (children's author)",
 'Author! Author! (film)',
 "Margaret Mayo (children's author)",
 "Al Perkins (children's author)",
 "John Lawson (children's author)",
 'Roald Dahl']

In [4]:
lit = wikipedia.page("Children's literature")

We can see what other pages the _Children's literature_ page has in it by checking out links. Conveniently these are the names of wikipedia pages.

In [30]:
lit.links[1:15]

['ALA Editions',
 'A Book of Giants',
 'A Little Pretty Pocket-Book',
 'Abanindranath Tagore',
 'Acronym and initialism',
 'Adventure book',
 'Adventurer',
 'Africa',
 'African American',
 'After the First Death',
 'Alan Garner',
 'Aleksandr Afanasyev',
 'Alex Rider',
 'Alexander Belyayev']

In [31]:
print(len(lit.links))

698


## Let The Scraping Begin!
Now that we have a list of pages that will get us what we want, we'll try to scrap all the content.
Scraping can introduce a burden on the website we are scraping. If you hit the website with too many requests in too short a time, it can cause websites slowdowns. We'll set the rate limit here so there is a breif pause before it tries to grab the data.

In [32]:
%%time
wikipedia.set_rate_limiting(True)

Wall time: 0 ns


698 pages should fit into memory pretty easily. if you are scraping a bunch, you'll need to periodically write the results to disk so you don't run out of memory.

In [35]:
%%time
bunchOfPages = [wikipedia.page(p).content for p in lit.links]

PageError: Page id "behari lal pura" does not match any pages. Try another id!

In [34]:
len(bunchOfPages)

10

In [23]:
import json
with open('littleList.json', 'w') as outfile:
    json.dump(BunchOfPages, outfile)

In [24]:
with open('littleList.json') as infile:
    loaded = json.load(infile)

'Alan Alexander Milne (; 18 January 1882 – 31 January 1956) was a British author, best known for his books about the teddy bear Winnie-the-Pooh and for various poems. Milne was a noted writer, primarily as a playwright, before the huge success of Pooh overshadowed all his previous work. Milne served in both World Wars, joining the British Army in World War I, and was a captain of the British Home Guard in World War II.\n\n\n== Biography ==\nAlan Alexander Milne was born in Kilburn, London to parents John Vine Milne, who was born in Jamaica,  and Sarah Marie Milne (née Heginbotham) and grew up at Henley House School, 6/7 Mortimer Road (now Crescent), Kilburn, a small public school run by his father. One of his teachers was H. G. Wells, who taught there in 1889–90.  Milne attended Westminster School and Trinity College, Cambridge  where he studied on a mathematics scholarship, graduating with a B.A. in Mathematics in 1903.  He edited and wrote for Granta, a student magazine.  He collabor